components/babel.config.js

```diff
@@ -7,7 +7,8 @@ module.exports = {
         useBuiltIns: "entry"
       }
     ],
-    "@babel/preset-react"
+    "@babel/preset-react",
+    "@babel/preset-typescript"
```

Comment:
Hmm...looks like you already use Babel's typescript preset, so you really don't need `ts-loader`.